### ISSUE TITLE
Handle action creators that return a function

### ISSUE THREAD

**Issue by vincent** [hardchor/electron-redux#9]
Hi,

(comming from your timesheets project as a great starting point for my app, so thanks for that)

When trying to make action creators that return a function, as described in [redux-thunk docs](https://github.com/gaearon/redux-thunk#motivation), we get an error

> Cannot read property 'substr' of undefined

I'm new to react/redux so maybe there's something I missed, but I think it is caused by https://github.com/hardchor/electron-redux/blob/master/src/middleware/forwardToMain.js#L4 where it does not check action's type. 

I think it should ignore if it's a function, doesn't it ?

**Comment by hardchor** [comment]
Hey,

Glad the timesheets app helped as a starting point!

You're absolutely right, the error comes from [here](https://github.com/hardchor/electron-redux/blob/2e6374434f3979a1996b0485ca9647df2e833a72/src/middleware/forwardToMain.js#L5).

I'll get a fix together tonight, unless you want to open a PR? 😄 

Have you had a look at https://github.com/hardchor/electron-redux#aliased-actions-main-process yet?

**Comment by hardchor** [comment]
Note to self:
* add check `if action.type` for BC
* update docs to run `forwardToMain` last, not first

**Comment by vincent** [comment]
Glad I could help a bit.

I'll let you do the change, as you may want to do other things, I'm not in a hurry anyway.

Yes, I did played with `createAliasedAction`. I think I get when to use it, but I still need to experiment  some aspects. If you'd like remarks from a react noob on that repo I could find some :)

Salut !

**Comment by hardchor** [comment]
Hey @vincent 

I've just published `v1.2.1` which addresses a few things:

[`redux-thunk` actions](https://github.com/gaearon/redux-thunk#motivation) (`incrementAsync`) aren't FSA-compliant to start with. However, the resulting triggered action (`increment`) _can_ be.
So, to cater for your case, I've added a check in there that ignores any non-compliant actions, so that your thunk actions should still work. 

Bear in mind, however, that any async action will be executed in the renderer process. This might not always be what you want, e.g. when you're fetching data.
The preferred way is to use `createAliasedAction`, so that those are being forwarded and executed in the main store.

I've created an extensive example in https://github.com/hardchor/timesheets/tree/thunk-actions%239 and left you some comments under https://github.com/hardchor/timesheets/pull/211/files#diff-1436cd21d24f5e8d073ff03c8384fc4a to explain what each action does. Check out the branch to see them in action on [Home](https://github.com/hardchor/timesheets/blob/thunk-actions%239/app/renderer/main/components/Home/index.jsx)
